File: Medium/yetAnother/yetAnother.c
#include <stdio.h>
#include <assert.h>

int main(void){
	int T;
	long long int R,B,P,ans;

	scanf("%d",&T);
	assert(T>=1 && T<=10);

	while(T>0){
		scanf("%lld %lld %lld",&R,&B,&P);

		assert(R>=1 && R<=100000);
		assert(B>=1 && B<=100000);
		assert(P>=0 && P<=R+B);

		ans = R+B-P;
		printf("%.12lf\n",(double) (R*P+B*ans)/(R+B));

		T--;
	}
	
	return 0;
}

File: Medium/yetAnother/yetAnother.java
//run as java -ea yetAnother
import java.util.Scanner;

public class yetAnother{
	public static void main(String[] args){
		int T;
		long R,B,P,ans;
		Scanner scan = new Scanner(System.in);

		T = scan.nextInt();
		assert(T>=1 && T<=10) : "T must be 1 <= T <= 10";

		while(T>0){
			R = scan.nextLong();
			B = scan.nextLong();
			P = scan.nextLong();

			assert(R>=1 && R<=100000) : "R must be 1 <= R <= 100000";
			assert(B>=1 && B<=100000) : "B must be 1 <= B <= 100000";
			assert(P>=0 && P<=R+B) : "P must be 0 <= P <= R+B";

			ans = R+B-P;
			
			System.out.printf("%.12f",(double) (R*P+B*ans)/(R+B));
			System.out.println();

			T--;
		}
	}
}